src/main/kotlin/it/pagopa/checkout/authservice/services/AuthLoginService.kt

```diff
@@ -0,0 +1,21 @@
+package it.pagopa.checkout.authservice.services
+
+import it.pagopa.checkout.authservice.clients.oneidentity.OneIdentityClient
+import it.pagopa.checkout.authservice.exception.OneIdentityClientException
+import it.pagopa.generated.checkout.authservice.v1.model.LoginResponseDto
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+
+@Service
+class AuthLoginService(private val oneIdentityClient: OneIdentityClient) {
+
+    fun login(): Mono<LoginResponseDto> {
+
+        return oneIdentityClient.buildLoginUrl().map { url ->
+            if (url.isBlank()) {
+                throw OneIdentityClientException("Generated URL is empty")
+            }
+            LoginResponseDto().urlRedirect(url)
+        }
+    }
```

Comment:
```suggestion
    fun login(): Mono<LoginResponseDto> {
        return oneIdentityClient.buildLoginUrl().map { 
            LoginResponseDto().urlRedirect(it)
        }
    }
```

```suggestion
    fun login(): Mono<LoginResponseDto> {

        return oneIdentityClient.buildLoginUrl().map { url ->
            if (url.isBlank()) {
                throw OneIdentityClientException("Generated URL is empty")
            }
            LoginResponseDto().urlRedirect(url)
        }
    }
```

redundant check here, check already performed in the called method.

Reply:
Good catch, addressed with [b3eeae3](https://github.com/pagopa/pagopa-checkout-auth-service/pull/12/commits/b3eeae3bf466ecd72d0aab25cccdea69b15d0839) ✅